src/MongoDB/ReadPreference.c

```diff
@@ -264,6 +312,38 @@ static PHP_METHOD(ReadPreference, __set_state)
 	php_phongo_readpreference_init_from_hash(intern, props);
 } /* }}} */
 
+/* {{{ proto array|null MongoDB\Driver\ReadPreference::getHedge()
+   Returns the ReadPreference hedge document */
+static PHP_METHOD(ReadPreference, getHedge)
+{
+	php_phongo_readpreference_t* intern;
+	const bson_t*                hedge;
+
+	intern = Z_READPREFERENCE_OBJ_P(getThis());
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		return;
+	}
+
+	hedge = mongoc_read_prefs_get_hedge(intern->read_preference);
+
+	if (!bson_empty0(hedge)) {
+		php_phongo_bson_state state;
+
+		PHONGO_BSON_INIT_STATE(state);
+		state.map.root_type = PHONGO_TYPEMAP_NATIVE_ARRAY;
+
+		if (!php_phongo_bson_to_zval_ex(bson_get_data(hedge), hedge->len, &state)) {
+			zval_ptr_dtor(&state.zchild);
```

Comment:
Exception is already thrown in this case, correct?

Reply:
`php_phongo_bson_to_zval_ex` throws an exception and returns `false` on error.